command/src/state.rs

```diff
@@ -12,51 +11,38 @@ use std::{
 use anyhow::{bail, Context};
 
 use crate::{
-    certificate::{calculate_fingerprint, CertificateAndKey, Fingerprint},
-    request::{
-        ActivateListener, AddBackend, AddCertificate, Cluster, DeactivateListener, ListenerType,
-        RemoveBackend, RemoveCertificate, RemoveListener, ReplaceCertificate, Request,
-        RequestHttpFrontend, RequestTcpFrontend,
-    },
-    response::{
-        Backend, ClusterInformation, HttpFrontend, HttpListenerConfig, HttpsListenerConfig,
-        PathRule, TcpFrontend, TcpListenerConfig,
+    certificate::{calculate_fingerprint, Fingerprint},
+    proto::command::{
+        ActivateListener, AddBackend, AddCertificate, CertificateAndKey, Cluster,
+        DeactivateListener, HttpListenerConfig, HttpsListenerConfig, ListenerType, PathRule,
+        RemoveBackend, RemoveCertificate, RemoveListener, ReplaceCertificate, RequestHttpFrontend,
+        RequestTcpFrontend, TcpListenerConfig,
     },
+    request::Request,
+    response::{Backend, ClusterInformation, HttpFrontend, TcpFrontend},
 };
 
 /// To use throughout Sōzu
 pub type ClusterId = String;
 
-#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
-pub struct HttpProxy {
-    address: SocketAddr,
-    fronts: HashMap<ClusterId, Vec<HttpFrontend>>,
-    backends: HashMap<ClusterId, Vec<Backend>>,
-}
-
-#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
-pub struct HttpsProxy {
-    address: SocketAddr,
-    certificates: HashMap<Fingerprint, CertificateAndKey>,
-    fronts: HashMap<ClusterId, Vec<HttpFrontend>>,
-    backends: HashMap<ClusterId, Vec<Backend>>,
-}
-
+// TODO: HttpFrontend and TcpFrontend and Backend are not present or meant to be
```

Comment:
Is always the case?

Reply:
it is still the case. For now, we have these types in protobuf:

- `RequestHttpFrontend`
- `RequestTcpFrontend`
- `AddBackend`
- `RemoveBackend`

these types contain `String` for socket addresses.

But the ConfigState, meant to be written in protobuf, contains these types:

- `HttpFrontend`
- `TcpFrontend`
- `Backend`

which contain field of the type `SocketAddress`, unwriteable in protobuf. We have to find a fix.

Reply:
Ok, it is not an protobuf issue, they are internal structures on which we use parsed structures. Do we have `TryFrom` trait implemented on them?

Like:
- `impl TryFrom<RequestHttpFrontend> for HttpFrontend`
- `impl TryFrom<RequestTcpFrontend> for TcpFrontend`
- `impl TryFrom<AddBackend> for Backend`
- `impl TryFrom<RemoveBackend> for Backend`

If we have them, we could merge this pull request and remove this comment as it something that we want.

Reply:
I don't see what you mean. For now, we do *receive* the `RequestxxxFrontend`, and `AddBackend` and `RemoveBackend` and parse them (albeit not with beautiful `TryFrom`).

The challenge here is that `ConfigState` is a structure meant to be *returned* when answering a `dump state` or a `save state` request.
Since `ConfigState` is returned, it can not contain internal `HttpFrontend`, `TcpFrontend` and `Backend`, which are not writeable in protobuf. It has to contain something else.